### DISCUSSION TITLE
Issues using the Library

### DISCUSSION THREAD

**Discussion by ManneN1** [nesrak1/AssetsTools.NET#43] [Q&A]
Hello there! Thanks for the amazing work on the library (and AssetView).

I've been trying to use it but I can't seem to get it to work as I'd like. After having imported the library with the package manager (Install-Package AssetsTools.NET -Version 2.0.9) and running the following code I get a, seemingly, empty bundle - even though I know it contains data (I can view it in AssetView just fine).

```
string path = Path.Combine(Environment.CurrentDirectory, RelativePath);

AssetsManager am = new AssetsManager();

FileInfo file = new FileInfo("classdata.tpk");
        
if (file.Exists)
{
    Console.WriteLine("The TPK file exists.");
}

am.LoadClassPackage("classdata.tpk");

file = new FileInfo(path);

if (file.Exists)
{
    Console.WriteLine("Bundle file exists.");
}

BundleFileInstance bundle = am.LoadBundleFile(path, false);

if (bundle.assetsFiles.Count > 0)
{
    Console.WriteLine("Found files in bundle!");
}
```

As you might expect, given that I'm writing this, I'm only getting the two first prints ("The TPK file exists" and "Bundle file exists").

I had a brief look also in the code of AssetView just to see how it was done there, and it seems to be the exact same. It seems to not matter in which order I perform the .LoadClassPackage() and the .LoadBundleFile() calls - still the same result, bundle.assetsFiles is 0.

I feel as though I'm missing something important. Got any idea what might be wrong?

PS: Attaching a sample bundle that shows up in AssetView and the .tpk file I'm using (which I copied over from the AssetView folder).

[data.zip](https://github.com/nesrak1/AssetsTools.NET/files/6919877/data.zip)

**Comment by nesrak1** [comment]
Use `LoadAssetsFileFromBundle` (see readme) instead of `bundle.assetsFiles`. I've already changed this to `loadedAssetsFiles` in the commit I'm about to make since that's really what it is. I think it was added only recently to support unloading assets files with their bundle.

**Reply by ManneN1** [reply]
Thanks for the fast response! 

I misunderstood the readme. I thought LoadAssetsFileFromBundle() was just a method for getting an asset by index (rather than iterating through the list, not sure why that'd be needed as you could just use [] through the list).

User error (as per usual) :)

Got it sorted now.
